lib/test.py

```diff
@@ -83,6 +75,10 @@ def write(self, fragment):
                 self.stream.write(line)
 
     def push_filter(self, pattern, replacement):
+        if sys.version[0] == '3' and isinstance(pattern, str):
+            pattern = pattern.encode('utf-8')
+        if sys.version[0] == '3' and isinstance(replacement, str):
+            replacement = replacement.encode('utf-8')
```

Comment:
As I see, we always receive an `str` here as for `pattern` as well as for `replacement`. I verified it on the whole tarantool test collection. It seems, the patch was created for some intermediate version of the patchset, when it was not so. That means that we can remove `isinstance()` checks. If we'll introduce the `as_bytes()` helper, it'll look even more nice:

```python
def push_filter(self, pattern, replacement):
    self.filters.append([as_bytes(pattern), as_bytes(replacement)])
```

Reply:
The change is part of the 'python3: decouple bytes and strings' commit now.